eth/utils/bls.py

```diff
@@ -170,15 +170,18 @@ def privtopub(k: int) -> int:
 
 
 def verify(message: bytes, pubkey: int, signature: bytes, domain: int) -> bool:
-    final_exponentiation = final_exponentiate(
-        pairing(FQP_point_to_FQ2_point(decompress_G2(signature)), G1, False) *
-        pairing(
-            FQP_point_to_FQ2_point(hash_to_G2(message, domain)),
-            neg(decompress_G1(pubkey)),
-            False
+    try:
+        final_exponentiation = final_exponentiate(
+            pairing(FQP_point_to_FQ2_point(decompress_G2(signature)), G1, False) *
+            pairing(
+                FQP_point_to_FQ2_point(hash_to_G2(message, domain)),
+                neg(decompress_G1(pubkey)),
+                False
```

Comment:
I find positional booleans to be confusing since they give no context on what is being `False`'d..  Maybe convert to either a keyword argument, or just use a comment to denote what argument this is for if performance is an issue.